bahmni-lite/docker-compose.yml

```diff
@@ -19,7 +19,9 @@ services:
 
   proxy:
     image: 'bahmni/proxy:${PROXY_IMAGE_TAG:?}'
-    #volumes:
+    volumes:
+      - "/etc/timezone:/etc/timezone:ro"
+      - "/etc/localtime:/etc/localtime:ro"
```

Comment:
This would make our docker compose bind to linux based systems. This would not work on Windows based machines/

Reply:
Should this come from .env file? And, we can inform users to comment it out for Windows systems?

Reply:
https://stackoverflow.com/a/3118775

As @mohan-13 rightly pointed out, this fix works for Debian-based Linux Distributions (Ubuntu, Arch Linux) and some Redhat-based systems (some versions of CentOS, RHEL, Fedora). On Debian-based systems, /etc/timezone is a file containing the timezone for that machine, and on some Redhat-based systems, you can get the required information from the /etc/localtime, which is a symlink. My approach was to combine both methods: use /etc/timezone if it exists; otherwise, try /etc/localtime. But this doesn't help if your host machine is in Windows. 

For Windows-based machines, the only reliable approach would be to make use of the TZ environment variable. The dependency of the TZ variable on the `tzdata` package means that we will have to have the package installed on all our containers. One way to get this done is to install the required package by running a shell command when starting the container, thus making sure the TZ env variable works without fail. 

`command: sh -c "if ! dpkg -s tzdata >/dev/null 2>&1; then apk update && apk add tzdata; fi && tail -f /dev/null"`
<img width="799" alt="Screenshot 2023-05-24 at 5 05 08 PM" src="https://github.com/Bahmni/bahmni-docker/assets/121226043/018e6931-dab9-4267-ae77-4e745268e063">

Here, the command first checks if the `tzdata` package is already installed using `dpkg -s tzdata`. If the package is not found, it proceeds to update the package lists with `apk update` and installs the `tzdata` package with `apk add tzdata`.  I noticed that the container immediately stops once this command alone is executed, so I have added the `tail -f /dev/null` command which is often used as a workaround or placeholder in Unix-like systems when you want a command that runs indefinitely without doing anything. The command `tail -f /dev/null` instructs tail to continuously monitor the /dev/null file for new lines, even though `/dev/null` doesn't contain any actual content. The result is that the command runs indefinitely without consuming system resources. This approach works except for services like bahmni-web and crater-php. But by running the `tail -f /dev/null`, I also suspect we might keep other tasks from executing. 

**Though having TZ environment variable gives us more flexibility, it adds the unnecessary overhead of having to install the `tzdata` package. I would suggest making use of the /etc/timezone, /etc/localtime in read-only mode instead.**

Reply:
In our PAT call (Wed, 24 May 2023) it was decided that we we will install the `tzdata` package in all the required images (proxy and crater-php) by directly running an install command in the `Dockerfile`, and make use of TZ environment variable instead of syncing the host’s timezone by mounting the relevant files into your containers. As @umair-fayaz suggested, the use of TZ environment variable gives us more flexibility and makes it easy to configure the container timezone independent of the host machine OS.